### ISSUE TITLE
Webform JS and CSS not loading properly

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/ckeditor_blocks#25]
If you save a webform block into a node it appears fine after saving, but if you reload the page the webform looses it's formatting. We have verified that the CSS and JS files provided by the webform module are not being loaded. 

If you clear the cache, the page and webform appear as expected, but again if you reload the page without clearing the cache, the JS and CSS fail to load and the webform looses it's formatting.

We have temporarily implemented a brute force fix that loads these specific JS and CSS files on every node. We could make this load conditional on pages with Webform blocks in the body field, but I am not sure if that is the best solution. 

```
function ckeditor_blocks_node_view(Node $node, $view_mode, $langcode) {
  backdrop_add_js('modules/webform/js/webform.js');
  backdrop_add_css('modules/webform/css/webform.css');  
}
```

We are looking for ideas. 

Exhibit A - What the form looks like after saving node or clearing cache

![image](https://user-images.githubusercontent.com/3144571/157003938-90f7fe86-e49b-4c06-b08c-5b066d351fb9.png)

Exhibit B - What the form looks like after reloading the page, without clearing cache. 

![image](https://user-images.githubusercontent.com/3144571/157004096-f02f4fe5-bc5f-4c84-87bf-3a1d552d5b09.png)

**Comment by stpaultim** [comment]
@hosef - If I understood you correctly the other day, you suggested one of two options.

1) That the problem might be that the block is only rendering on node save and that we need to force the node/block to render on every load. 

This sounded good, but when looking at our module it's not clear to me how this would be implemented or if it is actually an accurate description of the problem. 

2) I believe that you also suggested that there might be a way to turn the rendering of these blocks over to the token module, which might do a better job of it. 

Did I understand your suggestions correctly and/or can you improve my understanding of them. 

Flagging @rajpaul who is working with me on this project.

**Comment by stpaultim** [comment]
Last note: My colleague @rajpaul believes that the problem is related to the filter and the caching of the filter. @rajpaul, please feel free to elaborate.

**Comment by stpaultim** [comment]
@rajpaul - There might be something useful in this recent Zulip discussion about custom tokens.

https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Custom.20Tokens/near/274400918

**Comment by rajpaul** [comment]
I will take a look @stpaultim

**Comment by stpaultim** [comment]
@rajpaul - In Backdrop CMS Office Hours today I chatted with @hosef about this issue again. Here is what we came up with. 

1) The problem seems to be that entity caching is storing the webform and using the cached version upon page reload, but entity caching does not cache the webform specific css and js. 

2) My brute force solution does not work, because it leaves the webform in entity cache and the webform must be rebuilt each time, because of some special codes that MUST be regenerated. A cached webform will not work properly, even if we force load the JS and CSS.

3) Cache Tags - if implemented in Backdrop CORE might solve this problem. Although, given the issue with the regenerated codes, it's not clear to me that it will solve this problem. 

4) My new brute force solution suggested by @hosef and @argiepiano is to turn off entity caching for all nodes. This seems to solve the problem, but might have significant performance implications. I am considering a module setting that allows the site admin to change a setting for this module that either enables or disables entity caching. 

```
function ckeditor_blocks_entity_info_alter(&$info) {
  $info['node']['entity cache'] = FALSE;
}
```

I am inclined to consider this issue closed, but I will do a little more testing first.
